src/main/java/server/CommandRegistry.java

```diff
@@ -0,0 +1,58 @@
+package server;
+
+import server.commands.Command;
+import server.commands.Commands;
+import server.commands.StatusResponder;
+import server.handlers.FileHandler;
+
+import java.io.IOException;
+
+public class CommandRegistry {
+
+    private FileHandler fileHandler;
+    private StatusResponder statusResponder;
+    private Commands commands;
+
+    public CommandRegistry(FileHandler fileHandler, StatusResponder statusResponder) {
+        this.fileHandler = fileHandler;
+        this.statusResponder = statusResponder;
+        this.commands = new Commands(this::unrecognized);
+        registerCommands();
+    }
+
+    public void executeCommand(String commandName, String argument) throws IOException {
+        commands.execute(commandName, argument);
+    }
+
+    private void registerCommands() {
+        commands.register(
+                new Command("RETR", this::RETR),
+                new Command("STOR", this::STOR)
+        );
+    }
+
+    private void STOR(String fileName) {
+        try {
+            statusResponder.respond(150, "OK receiving file");
+            fileHandler.store(fileName);
+            statusResponder.respond(250, "OK File stored");
+        } catch (IOException e) {
+            statusResponder.respond(450, "Something went wrong");
```

Comment:
`450 Blocked by Windows Parental Controls (Microsoft)`
First time I see it

Reply:
@fabientownsend lol I got `450` from here http://www.serv-u.com/kb/1522/450-FTP-Reply-Code, but I've yet to go through the formalised response codes yet (just getting it working for now)

Reply:
OH right, FTP protocol, nevermind! :D